xcarchive/xcarchive.go

```diff
@@ -39,3 +42,38 @@ func IsMacOS(archPath string) (bool, error) {
 
 	return exist, nil
 }
+
+func unwrapFileEmbeddedInXcarchiveDir(xcarchivePth, fileName string) (string, error) {
```

Comment:
I think this is really not common to have only one file from an archive, and with this solution, each time you are looking for a file will create a new temp dir and unzip the archive. Try to find a solution where you can give an array of files in and then receive the files back somehow. So you can skip the extracting step each time.

Reply:
Updated code, used different approach.